src/sunmatrix/band/sunmatrix_band.c

```diff
@@ -375,23 +368,20 @@ int SUNMatScaleAdd_Band(realtype c, SUNMatrix A, SUNMatrix B)
     for (i=-SM_UBAND_B(B); i<=SM_LBAND_B(B); i++)
       A_colj[i] = c*A_colj[i] + B_colj[i];
   }
-  return SUNMAT_SUCCESS;
+  return SUN_SUCCESS;
 }
 
-int SUNMatMatvec_Band(SUNMatrix A, N_Vector x, N_Vector y)
+SUNErrCode SUNMatMatvec_Band(SUNMatrix A, N_Vector x, N_Vector y)
 {
+  SUNAssignSUNCTX(A->sunctx);
   sunindextype i, j, is, ie;
   realtype *col_j, *xd, *yd;
 
-  /* Verify that A, x and y are compatible */
-  if (!SMCompatible2_Band(A, x, y))
-    return SUNMAT_ILL_INPUT;
+  SUNCheck(compatibleMatrixAndVectors(A, x, y), SUN_ERR_ARG_DIMSMISMATCH);
 
   /* access vector data (return if failure) */
-  xd = N_VGetArrayPointer(x);
-  yd = N_VGetArrayPointer(y);
-  if ((xd == NULL) || (yd == NULL) || (xd == yd))
-    return SUNMAT_MEM_FAIL;
+  xd = SUNCheckCallLastErr(N_VGetArrayPointer(x));
+  yd = SUNCheckCallLastErr(N_VGetArrayPointer(y));
```

Comment:
This no longer performs a check to ensure that `xd` and `yd` point to different memory addresses.  I recommend that this add the line:
`SUNAssert(xd != yd, SUN_ERR_ARG_INCOMPATIBLE);`